File: freecodecamp/daily-code-challenge/2026-03-31.js
function alarmCheck(alarmTime, wakeTime) {
  const toMinutes = (time) => {
    const [h, m] = time.split(":").map(Number);
    return h * 60 + m;
  };

  const alarm = toMinutes(alarmTime);
  const wake = toMinutes(wakeTime);

  if (wake < alarm) return "early";
  if (wake <= alarm + 10) return "on time";
  return "late";
}

File: freecodecamp/daily-code-challenge/2026-03-31.py
def alarm_check(alarm_time, wake_time):
    def to_minutes(time):
        h, m = map(int, time.split(":"))
        return h * 60 + m

    alarm = to_minutes(alarm_time)
    wake = to_minutes(wake_time)

    if wake < alarm:
        return "early"
    elif wake <= alarm + 10:
        return "on time"
    else:
        return "late"